Edit snippets in modal

olmokramer opened the pull request:
Snippet replacement texts can be multiple lines, but the user interface didn't reflect this with its single line input. Changing it to a resizeable text area makes it much easier to use for longer replacements.

Closes #639.
Closes #659.
Closes #847.
Closes #850.
Closes #860.

libre-man commented on src/components/SnippetManager.vue:
```diff
@@ -8,66 +8,86 @@
+                <th>
+                    <b-button size="sm"
+                              variant="primary"
+                              @click="editSnippet()"
```
The `()` are not needed here.

olmokramer replied:
No, they are required. Without them the first argument to the function would be the event.

libre-man commented on src/components/SnippetManager.vue:
```diff
@@ -8,66 +8,86 @@
+            <b-button-toolbar justify
+                              class="modal-buttons">
+                <b-button variant="danger"
+                          @click="cancelEditSnippet">
```
I think both should be a label or both should be a slot.

olmokramer replied:
Sure. `<b-button>` does not have a `label` prop, so I'll make them both a slot.

olmokramer commented:
> After submitting using Ctrl-RET I was presented with this modal: . The same happens when I press "Save".

Fixed this.

> The dialog in the sidebar needs some bottom margin imo.

Hmm, strange that there's no margin in firefox. There is in chromium.

> I think a description popover explaining snippets and how to use them would be useful.

Yes that's a good idea. I've added one.

> The plus in the add button seems to be placed a bit to high vertically.

Yep, I've fixed this. This can't really be fixed. As the image below shows, icons are not always centered in their bounding box. I've changed the defaults a bit so it looks nice for most icons, but we will have to fix up the icons like the plus signs individually.

![x](https://user-images.githubusercontent.com/3612514/54641382-5f88eb00-4a92-11e9-8ee8-eeb1dc1f48a1.png)

I also noticed in your screenshot that the sidebar seems to have a higher z-index than the modal backdrop, so I've also fixed that. It's a bit of a hack, drawing a box with exactly the same size as the main menu and the submenus and setting their z-index, so the z-indices of the `.main-menu` and the `.submenu-container` could be equal...

olmokramer commented:
I've changed the behaviour a bit as we discussed when you enter a duplicate key. It now shows a confirmation popover on the save button. If the user clicks "yes" the snippet with the duplicate key will be overwritten with the one being edited. If the latter one already existed in the database it is deleted.

libre-man commented:
The edit buttons need to be reversed.

olmokramer commented:
I've reversed the buttons, and also increased their size, because IMO it looks better than the small buttons.
